source/common/router/shadow_writer_impl.cc

```diff
@@ -11,31 +11,57 @@
 namespace Envoy {
 namespace Router {
 
+namespace {
+
+std::string newHost(absl::string_view host) {
```

Comment:
Maybe give this a name that better describes what it does? An example could be helpful as well